syft/source/directory_resolver.go

```diff
@@ -171,6 +171,13 @@ func (r directoryResolver) addPathToIndex(p string, info os.FileInfo) (string, e
 		if err != nil {
 			return "", fmt.Errorf("unable to readlink for path=%q: %w", p, err)
 		}
+
+		for _, filterFn := range r.pathFilterFns {
```

Comment:
Filtering should occur within `indexPath`. If there was a need to put filtering logic here it may mean that this is a usage of `addPathToIndex` where there should be an `indexPath`

Reply:
I misread this.